bitcoin/src/blockdata/script/push_bytes.rs

```diff
@@ -0,0 +1,463 @@
+//! Contains `PushBytes` & co
+
+use core::ops::{Deref, DerefMut};
+use core::borrow::{Borrow, BorrowMut};
+#[allow(unused)]
+use crate::prelude::*;
+
+pub use primitive::*;
+
+/// This module only contains required operations so that outside functions wouldn't accidentally
+/// break invariants. Therefore auditing this module should be sufficient.
+mod primitive {
+    #[allow(unused)]
+    use crate::prelude::*;
+
+    use super::PushBytesError;
+    use core::convert::{TryFrom, TryInto};
+    use core::ops::{Index, Range, RangeFull, RangeFrom, RangeTo, RangeInclusive, RangeToInclusive};
+    #[cfg(feature = "rust_v_1_53")]
+    use core::ops::Bound;
+
+    #[cfg(any(target_pointer_width = "16", target_pointer_width = "32"))]
+    fn check_limit(len: usize) -> Result<(), PushBytesError> {
+        Ok(())
+    }
+
+    #[cfg(not(any(target_pointer_width = "16", target_pointer_width = "32")))]
```

Comment:
In bcd95fa0368c12c3cc8e01bdc896cc738054b648:

Maybe a comment here saying "we have 64 bits in mind, but even for esoteric sizes, this code is correct, since it's the conservative one that checks for errors".